sparql/SPARQLParser.g4

```diff
@@ -0,0 +1,694 @@
+ | CEIL OPAR expression CPAR
+ | FLOOR OPAR expression CPAR
+ | ROUND OPAR expression CPAR
+ | CONCAT expressionList
+ | substringExpression
+ | STRLEN OPAR expression CPAR
+ | strReplaceExpression
+ | UCASE OPAR expression CPAR
+ | LCASE OPAR expression CPAR
+ | ENCODE_FOR_URI OPAR expression CPAR
+ | CONTAINS OPAR expression COMMA expression CPAR
+ | STRSTARTS OPAR expression COMMA expression CPAR
+ | STRENDS OPAR expression COMMA expression CPAR
+ | STRBEFORE OPAR expression COMMA expression CPAR
+ | STRAFTER OPAR expression COMMA expression CPAR
+ | YEAR OPAR expression CPAR
+ | MONTH OPAR expression CPAR
+ | DAY OPAR expression CPAR
+ | HOURS OPAR expression CPAR
+ | MINUTES OPAR expression CPAR
+ | SECONDS OPAR expression CPAR
+ | TIMEZONE OPAR expression CPAR
+ | TZ OPAR expression CPAR
+ | NOW NIL
+ | UUID NIL
+ | STRUUID NIL
+ | MD5 OPAR expression CPAR
+ | SHA1 OPAR expression CPAR
+ | SHA256 OPAR expression CPAR
+ | SHA384 OPAR expression CPAR
+ | SHA512 OPAR expression CPAR
+ | COALESCE expressionList
+ | IF OPAR expression COMMA expression COMMA expression CPAR
+ | STRLANG OPAR expression COMMA expression CPAR
+ | STRDT OPAR expression COMMA expression CPAR
+ | SameTerm OPAR expression COMMA expression CPAR
+ | IsIRI OPAR expression CPAR
+ | IsURI OPAR expression CPAR
+ | IsBLANK OPAR expression CPAR
+ | IsLITERAL OPAR expression CPAR
+ | IsNUMERIC OPAR expression CPAR
+ | regexExpression
+ | existsFunc
+ | notExistsFunc
+ ;
+
+/// [122]    RegexExpression : 'REGEX' '(' Expression ',' Expression ( ',' Expression )? ')'
+regexExpression : REGEX OPAR expression COMMA expression ( COMMA expression )? CPAR;
+
+/// [123]    SubstringExpression : 'SUBSTR' '(' Expression ',' Expression ( ',' Expression )? ')'
+substringExpression : SUBSTR OPAR expression COMMA expression ( COMMA expression )? CPAR;
+
+/// [124]    StrReplaceExpression : 'REPLACE' '(' Expression ',' Expression ',' Expression ( ',' Expression )? ')'
+strReplaceExpression : REPLACE OPAR expression COMMA expression COMMA expression ( COMMA expression )? CPAR;
+
+/// [125]    ExistsFunc : 'EXISTS' GroupGraphPattern
+existsFunc : EXISTS groupGraphPattern;
+
+/// [126]    NotExistsFunc : 'NOT' 'EXISTS' GroupGraphPattern
+notExistsFunc : NOT EXISTS groupGraphPattern;
+
+/// [127]    Aggregate
+///           : 'COUNT' '(' 'DISTINCT'? ( '*' | Expression ) ')'
+///           | 'SUM' '(' 'DISTINCT'? Expression ')'
+///           | 'MIN' '(' 'DISTINCT'? Expression ')'
+///           | 'MAX' '(' 'DISTINCT'? Expression ')'
+///           | 'AVG' '(' 'DISTINCT'? Expression ')'
+///           | 'SAMPLE' '(' 'DISTINCT'? Expression ')'
+///           | 'GROUP_CONCAT' '(' 'DISTINCT'? Expression ( ';' 'SEPARATOR' '=' String )? ')'
+aggregate
+ : COUNT OPAR DISTINCT? ( MUL | expression ) CPAR
+ | SUM OPAR DISTINCT? expression CPAR
+ | MIN OPAR DISTINCT? expression CPAR
+ | MAX OPAR DISTINCT? expression CPAR
+ | AVG OPAR DISTINCT? expression CPAR
+ | SAMPLE OPAR DISTINCT? expression CPAR
```

Comment:
```suggestion
 | ( SUM | MIN | MAX | AVG | SAMPLE ) OPAR DISTINCT? expression CPAR
```

Reply:
In the future, I'll add parser labels to these alts, so here too, I don't want to rewrite things. Sorry.

Reply:
BTW, this simplification also doesn't change parse tree, because there is no ambiguity.

Reply:
No, but I will not be able to add alt-labels with this change. Again, no problem, I'll keep the grammar in my own repo. Really no big deal.